docs/rules/terraform_no_short_circuit_evaluation.md

```diff
@@ -0,0 +1,60 @@
+# terraform_no_short_circuit_evaluation
+
+Disallow using logical operators (`&&`, `||`) with null checks that could lead to errors due to lack of short-circuit evaluation.
```

Comment:
```suggestion
Detects usage of logical operators (`&&`, `||`) that could lead to errors because of their lack of support for short-circuit evaluation.
```

Doesn't seem like null is an essential part of the description, you might be looking at length or any other intended pre-condition.